packages/fx-core/src/component/driver/deploy/azure/azureFunctionDeployDriver.ts

```diff
@@ -22,63 +18,38 @@ export class AzureFunctionDeployDriver implements StepDriver {
   readonly description: string = getLocalizedString(
     "driver.deploy.deployToAzureFunctionsDescription"
   );
+  private static readonly SERVICE_NAME = "Azure Function App";
+  private static readonly SUMMARY = [
+    // eslint-disable-next-line no-secrets/no-secrets
+    getLocalizedString("driver.deploy.azureFunctionsDeploySummary"),
+  ];
+  private static readonly SUMMARY_PREPARE = [
+    getLocalizedString("driver.deploy.notice.deployDryRunComplete"),
+  ];
 
   @hooks([addStartAndEndTelemetry(ACTION_NAME, TelemetryConstant.DEPLOY_COMPONENT_NAME)])
   async run(args: unknown, context: DriverContext): Promise<Result<Map<string, string>, FxError>> {
-    const impl = new AzureFunctionDeployDriverImpl(args, context);
-    return wrapRun(
-      () => impl.run(),
-      () => impl.cleanup(),
-      context.logProvider
+    const impl = new AzureZipDeployDriverImpl(
+      args,
+      context,
+      AzureFunctionDeployDriver.SERVICE_NAME,
+      "https://aka.ms/teamsfx-actions/azure-functions-deploy",
+      AzureFunctionDeployDriver.SUMMARY,
+      AzureFunctionDeployDriver.SUMMARY_PREPARE
     );
+    return (await impl.run()).result;
   }
 
+  @hooks([addStartAndEndTelemetry(ACTION_NAME, TelemetryConstant.DEPLOY_COMPONENT_NAME)])
   async execute(args: unknown, ctx: DriverContext): Promise<ExecutionResult> {
-    const impl = new AzureFunctionDeployDriverImpl(args, ctx);
-    const res = await wrapRun(
-      () => impl.run(),
-      () => impl.cleanup(),
-      ctx.logProvider
-    );
-    if (impl.dryRun) {
-      return wrapSummary(async () => {
-        return res;
-      }, ["driver.deploy.notice.deployDryRunComplete"]);
-    }
-    return wrapSummary(async () => {
-      return res;
-      // eslint-disable-next-line no-secrets/no-secrets
-    }, ["driver.deploy.azureFunctionsDeploySummary"]);
-  }
-}
-
-/**
- * deploy to Azure Function
- */
-export class AzureFunctionDeployDriverImpl extends AzureDeployDriver {
-  pattern =
-    /\/subscriptions\/([^\/]*)\/resourceGroups\/([^\/]*)\/providers\/Microsoft.Web\/sites\/([^\/]*)/i;
-  protected helpLink = "https://aka.ms/teamsfx-actions/azure-functions-deploy";
-
-  async azureDeploy(
-    args: DeployStepArgs,
-    azureResource: AzureResourceInfo,
-    azureCredential: TokenCredential
-  ): Promise<void> {
-    await this.progressBar?.start();
-    await this.zipDeploy(args, azureResource, azureCredential);
-    if (!this.dryRun) {
-      await this.progressBar?.next(ProgressMessages.restartAzureFunctionApp);
-      await this.restartFunctionApp(azureResource);
-    }
-    await this.progressBar?.end(true);
-  }
-
-  createProgressBar(ui?: UserInteraction): IProgressHandler | undefined {
-    const steps = this.dryRun ? 1 : 6;
-    return ui?.createProgressBar(
-      `Deploying ${this.workingDirectory ?? ""} to Azure Function App`,
-      steps
+    const impl = new AzureZipDeployDriverImpl(
+      args,
+      ctx,
```

Comment:
Naming convention is better to be consistent. Sometimes it is `ctx` and sometimes it is `context`.

Reply:
+1